execute/tokendata/observer.go

```diff
@@ -116,18 +116,41 @@ func (c *compositeTokenDataObserver) Observe(
 	msgObservations exectypes.MessageObservations,
 ) (exectypes.TokenDataObservations, error) {
 	tokenDataObservations := c.initTokenDataObservations(msgObservations)
+	// Terminate immediately if there are no observers
+	if len(c.observers) == 0 {
+		return tokenDataObservations, nil
+	}
+
+	resultsCh := make(chan exectypes.TokenDataObservations, len(c.observers))
```

Comment:
Some thoughts about the general approach.

1. I'd like to see this be asynchronous to the Observe function. (i.e. persistent result channel + go routine collecting results).
2. During Observe, send new/failed attestation requests out for observation and grab ones that are ready.
3. Is there a timeout on the context?
4. Channel size should be one or none: [one or none](https://github.com/uber-go/guide/blob/master/style.md#channel-size-is-one-or-none).